### PR TITLE
fix: passing null object to type array defaults to empty array (#249)

### PR DESCRIPTION
**Pull request by alemagio**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.

### CODE REVIEW

**Review by zekth — APPROVED** [review]
lgtm

**Review by RafaelGSS — COMMENTED** [review]
Not sure if this is unexpected behavior. The following snippet sent an empty array by default
```js
const schema = {
  type: 'object',
  properties: {
    arr: {
      type: 'array',
      items: {
        type: 'number'
      },
      default: [],
    }
  }
}

const stringify = fastJson(schema)
const valueStr = stringify({}) // {"arr":[]}
```

Although throwing an error doesn't look good.

**Comment by alemagio** [comment]
@RafaelGSS Added a unit test with your case.
I'm not having any error, am I missing somenthing?

**Comment by RafaelGSS** [comment]
No, I said the current implementation(master) seems to the expected behavior since `null` is sent. I'll wait for some contributors to land this.

Btw the last test case is not necessary, is already being tested [here](https://github.com/fastify/fast-json-stringify/blob/master/test/defaults.test.js#L214)

**Comment by alemagio** [comment]
> No, I said the current implementation(master) seems to have expected behavior since `null` is sent. I'll wait for some contributors to give their opinion.
> 
> Btw the last test case is not necessary, is already being tested [here](https://github.com/fastify/fast-json-stringify/blob/master/test/defaults.test.js#L214)

Oh ok.
I'll remove last test in the meantime.

File: test/typesArray.test.js [hunk 1]
```diff
@@ -417,3 +417,22 @@ test('object that is simultaneously a string and a json switched', (t) => {
...
+    type: 'object',
+    properties: {
+      arr: {
+        type: 'array',
+        items: {
+          type: 'number'
+        }
+      }
+    }
+  }
```
Comment (mcollina) [inline test/typesArray.test.js:433]: Are you sure passing null for the array would validate this schema? Can you check with ajv?

My understanding is that null is not a valid array.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test/typesArray.test.js (hunk 1 above)
Comment (zekth) [reply test/typesArray.test.js:433]:
```javascript
const schema = {
  type: 'object',
  properties: {
    foo: {
      $id: '#/properties/foo',
      type: 'array',
      default: [],
      items: {
        $id: '#/properties/foo/items'
      }
    }
  }
}
const obj = {
  foo: null
}
var Ajv = require('ajv')
var ajv = new Ajv()
var validate = ajv.compile(schema)
var valid = validate(obj)
if (!valid) console.log(validate.errors)
// output
// [
//   {
//     keyword: 'type',
//     dataPath: '.foo',
//     schemaPath: '#/properties/foo/type',
//     params: { type: 'array' },
//     message: 'should be array'
//   }
// ]

```
Indeed

**Comment by alemagio** [comment]
I followed the issue, too strictly actually.
Tried with ajv and null is not valid when type is array.
I think a good way to handle this is to throw an error with a more informative message.
What do you think?

**Comment by mcollina** [comment]
> I think a good way to handle this is to throw an error with a more informative message.

Yes, go for it!

**Comment by alemagio** [comment]
Updated the code.
Now in case something expected to be an array is something else a clearer error is thrown.

**Comment by alemagio** [comment]
Would you mind adding `hacktoberfest-accepted` label to this PR so I can complete my Hacktoberfest? :smile:

**Comment by alemagio** [comment]
Thanks 😁

**Pull request merged**